### ISSUE TITLE
sequence-of not combinable

### ISSUE THREAD

**Issue by gregspurrier** [staples-sparx/clj-schema#1]
I wanted to add some constraints around the length of a sequence, so I tried this:

``` clojure
(defn- at-least
  [n]
  (fn [coll]
    (and (coll? coll)
         (>= (count coll) n))))

(defn- at-most
  [n]
  (fn [coll]
    (and (coll? coll)
         (<= (count coll) n))))

(def-loose-schema stuff
  [[:things] [(sequence-of string?)
              (at-least 1)
              (at-most 3)]])                 
```

I expected that the three constraints would be combined to mean a sequence of strings with 1 <= length <= 3.  But, here's what I get:

``` clojure
; user> (clj-schema.validation/validation-errors stuff {:things ["one"]})
#{"Map value [\"one\"], at path [:things], was sequential but not tagged with 'sequence-of'."}
```

If there's a preferred/supported way to support this use case, please let me know. Otherwise, it would be great if the above worked.

**Comment by CampingScorpion** [comment]
HI @gregspurrier,

There's a possibly-non-intuitive way to to do this.  

``` clj
(def-loose-schema stuff
  [[:things] [(sequence-of string?)
   [:things] [(at-least 1)
              (at-most 3)]])                 
```

**Comment by CampingScorpion** [comment]
I may give this some hammock time to consider ways to make it more intuitive to use.

**Comment by gregspurrier** [comment]
Thanks. I'll use the non-intuitive way for now.

**Comment by CampingScorpion** [comment]
I've been thinking that a way to fix this is to make it illegal for a path validated with a (sequence-of ...)  validator to be a single item, but ok for a single-item validator (the normal garden variety) to be applied to a sequence.

**Comment by CampingScorpion** [comment]
@gregspurrier Fixed this, so you should be able to use your original example for this case.  I've released it in clj-schema-0.7.0. I've also changed the ErrorReport protocol in this release.
